sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -967,12 +967,18 @@ object EliminateSorts extends Rule[LogicalPlan] {
  * Removes redundant Sort operation. This can happen:
  * 1) if the child is already sorted
  * 2) if there is another Sort operator separated by 0...n Project/Filter operators
+ * 3) if the Sort operator is within Join and without Limit
+ * 4) if the Sort operator is within GroupBy and the aggregate function is order irrelevant
  */
 object RemoveRedundantSorts extends Rule[LogicalPlan] {
```

Comment:
Can we rename it to `EliminateSorts`? Some sorts are not redundant but we remove them as well according to the SQL semantic.

Reply:
There is another rule named `EliminateSorts`. Can we merge it together?